picard/script/functions.py

```diff
@@ -1381,3 +1381,97 @@ def func_countryname(parser, country_code, translate=""):
     if translate:
         return gettext_countries(name)
     return name
+
+
+def _split_date(date_to_parse, date_order="ymd", separator="-"):
+    """Split the specified date into parts.
+
+    Args:
+        date_to_parse (str): Date string to parse
+        date_order (str, optional): Order of date elements. Can be "ymd", "mdy" or "dmy". Defaults to "ymd".
+        separator (str, optional): Separator between date parts. Defaults to "-".
+
+    Returns:
+        tuple: Tuple of the date parts as (year, month, day)
+    """
+    if separator == '':
+        separator = None
+    parts = date_to_parse.strip().split(separator)
+    parts.extend(['', '', ''])
+    date_order = date_order.lower()
+    if date_order == 'dmy':
+        return parts[2], parts[1], parts[0]
+    elif date_order == 'mdy':
+        return parts[2], parts[0], parts[1]
+    else:
+        return parts[0], parts[1], parts[2]
```

Comment:
We could use a `namedtuple` for the result.

```python
DateTuple = namedtuple('DateTuple', ('year', 'month', 'day'))

....

return DateTuple(parts[0], parts[1], parts[2])

```

So in caller functions, we can write:

```python
return _split_date(date_to_parse, date_order, separator).year
```